src/export-rs274x.c

```diff
@@ -301,29 +442,28 @@ gerbv_export_rs274x_file_from_image (const gchar *filename, gerbv_image_t *input
 		   the loaded file may refer to invalid apertures */
 		if ((currentNet->aperture != currentAperture)&&
 			(image->aperture[currentNet->aperture] != NULL)) {
-			fprintf(fd, "G54D%02d*\n",currentNet->aperture);
+			fprintf(fd, "%sD%02d*\n",
+			        x2 ? "" : "G54",        // G54 unnecessary in modern standards
+			        currentNet->aperture);
 			currentAperture = currentNet->aperture;
 		}
 		
 		oldLayer = currentNet->layer;
 		oldState = currentNet->state;
+
+	        if (x2 && !insidePolygon)
+	                x2attr_foreach_net_attr(currentNet, _dump_attribs, &xud);
 		
-		long xVal,yVal,endX,endY,centerX,centerY;
 		switch (currentNet->interpolation) {
 			case GERBV_INTERPOLATION_LINEARx1 :
 			case GERBV_INTERPOLATION_LINEARx10 :
 			case GERBV_INTERPOLATION_LINEARx01 :
 			case GERBV_INTERPOLATION_LINEARx001 :
 				/* see if we need to write an "aperture off" line to get
 				   the pen to the right start point */
-				if ((!insidePolygon) && (currentNet->aperture_state == GERBV_APERTURE_STATE_ON)) {
-					xVal = (long) round(currentNet->start_x * decimal_coeff);
-					yVal = (long) round(currentNet->start_y * decimal_coeff);
-					fprintf(fd, "G01X%07ldY%07ldD02*\n",xVal,yVal);
-				}
-				xVal = (long) round(currentNet->stop_x * decimal_coeff);
-				yVal = (long) round(currentNet->stop_y * decimal_coeff);
-				fprintf(fd, "G01X%07ldY%07ld",xVal,yVal);
+				if ((!insidePolygon) && (currentNet->aperture_state == GERBV_APERTURE_STATE_ON))
+					fprintf(fd, "G01%sD02*\n",_format_pair(&xud, currentNet->start_x, currentNet->start_y, FALSE));
+				fprintf(fd, "G01%s",_format_pair(&xud, currentNet->stop_x, currentNet->stop_y, FALSE));
```

Comment:
This is a great refactor and I am in favor of it but it could be in its own commit.